### ISSUE TITLE
Swagger 2.0.7 documenting classes that are not registered with jersey

### ISSUE THREAD

**Issue by josephlbarnett** [swagger-api/swagger-core#3145]
With the switch from reflections to classgraph, swagger is now documenting extra routes that are not actually registered with our jersey/dropwizard application.  It appears that classgraph is uncovering the following classes in JaxrsAnnotationScanner#classes(), where reflections previously did not pick them up:
-com.papertrail.profiler.jaxrs.CpuProfileResource
-org.glassfish.jersey.server.wadl.internal.WadlResource
-io.swagger.v3.jaxrs2.intergration.resources.AcceptHeaderOpenApiResource
-io.swagger.v3.jaxrs2.intergration.resources.OpenApiResource

the io.swagger resources appear to get ignored when building the documentation, but the dropwizard-transitive-dependency papertrail & glassfish ones are cluttering our swagger docs with unreachable endpoints.

We're initializing swagger as follows:
```
SwaggerContextBuilder()
            .openApiConfiguration(
                SwaggerConfiguration().openAPI(
                    OpenAPI().servers(listOf(Server().url("http://${appConfig.corsDomain}:${appConfig.appPort}/app")))
                )
            )
            .application(application)
            .ctxId(ctxId)
            .buildContext(true)
```

Basically, we'd like to only document resources that are actually available through the jersey application, not just resources that are available on the classpath.  Is this expected behavior?  Is there a way other than configuring the allowable packages (we want to filter based on the endpoint being reachable, not necessarily the author of the endpoint)?

**Comment by frantuma** [comment]
resolved resources depending on which `OpenApiContext` and `OpenApiScanner` implementations you are using, please check [wiki](https://github.com/swagger-api/swagger-core/wiki/Swagger-2.X---Integration-and-Configuration#resource-scanning) for details about provided scanners or providing your own implementation.

**Comment by josephlbarnett** [comment]
Thanks for the reply, looks like `JaxrsApplicationScanner` does what we want.  Was this an intended default behavior change or was it a bug in Reflections that `JaxrsApplicationAndAnnotationScanner` wasn't working properly before?

**Comment by frantuma** [comment]
I would say it was a bug in org.reflections (or a misconfiguration) not picking up for some reason resources in classpath
